test/parachain/tests/integration_test.rs

```diff
@@ -0,0 +1,404 @@
+
+		Ok(&self.stderr)
+	}
+
+	fn terminate(&mut self) {
+		match self.child.try_wait() {
+			Ok(Some(_)) => return,
+			Ok(None) => {}
+			Err(err) => {
+				eprintln!("could not wait for child process to finish: {}", err);
+				let _ = self.child.kill();
+				let _ = self.child.wait();
+				return;
+			}
+		}
+
+		#[cfg(unix)]
+		{
+			use nix::sys::signal::{kill, Signal::SIGTERM};
+			use nix::unistd::Pid;
+
+			kill(Pid::from_raw(self.child.id().try_into().unwrap()), SIGTERM).unwrap();
+
+			let mut tries = 30;
+
+			let success = loop {
+				tries -= 1;
+
+				match self.child.try_wait() {
+					Ok(Some(_)) => break true,
+					Ok(None) if tries == 0 => break false,
+					Ok(None) => thread::sleep(Duration::from_secs(1)),
+					Err(err) => {
+						eprintln!("could not wait for child process to finish: {}", err);
+						break false;
+					}
+				}
+			};
+
+			if !success {
+				let _ = self.child.kill();
+			}
+		}
+
+		#[cfg(not(unix))]
+		let _ = self.child.kill();
+
+		let _ = self.child.wait();
+	}
+}
+
+fn tcp_port_is_open<A: net::ToSocketAddrs>(address: A) -> bool {
+	match net::TcpStream::connect(&address) {
+		Ok(_) => true,
+		Err(_) => false,
+	}
+}
+
+fn wait_for_tcp<A: net::ToSocketAddrs>(address: A) -> io::Result<()> {
+	let mut tries = 10;
+
+	loop {
+		tries -= 1;
+
+		match net::TcpStream::connect(&address) {
+			Ok(_) => break Ok(()),
+			Err(err) if tries == 0 => break Err(err),
+			_ => thread::sleep(Duration::from_secs(1)),
+		}
+	}
+}
+
+#[async_std::test]
+#[ignore]
+async fn integration_test() {
+	assert!(
+		!tcp_port_is_open("127.0.0.1:9933"),
+		"tcp port is already open 127.0.0.1:9933, this test cannot be run",
+	);
+	assert!(
+		!tcp_port_is_open("127.0.0.1:9934"),
+		"tcp port is already open 127.0.0.1:9934, this test cannot be run",
+	);
+
+	// start alice
+	let polkadot_alice_dir = tempdir().unwrap();
+	let mut polkadot_alice = Command::new(cargo_bin("cumulus-test-parachain-collator"))
+		.stdout(Stdio::piped())
+		.stderr(Stdio::piped())
+		.args(POLKADOT_ARGS)
+		.arg("--base-path")
+		.arg(polkadot_alice_dir.path())
+		.arg("--alice")
+		.arg("--unsafe-rpc-expose")
+		.spawn()
+		.unwrap();
+	let polkadot_alice_helper = ChildHelper::new("alice", &mut polkadot_alice);
+	wait_for_tcp("127.0.0.1:9933").unwrap();
+
+	// start bob
+	let polkadot_bob_dir = tempdir().unwrap();
+	let mut polkadot_bob = Command::new(cargo_bin("cumulus-test-parachain-collator"))
+		.stdout(Stdio::piped())
+		.stderr(Stdio::piped())
+		.args(POLKADOT_ARGS)
+		.arg("--base-path")
+		.arg(polkadot_bob_dir.path())
+		.arg("--bob")
+		.arg("--unsafe-rpc-expose")
+		.arg("--rpc-port=9934")
+		.spawn()
+		.unwrap();
+	let polkadot_bob_helper = ChildHelper::new("bob", &mut polkadot_bob);
+	wait_for_tcp("127.0.0.1:9934").unwrap();
+
+	// wait a bit for some relay chains blocks to be generated
+	thread::sleep(Duration::from_secs(10));
+
+	// export genesis state
+	let cmd = Command::new(cargo_bin("cumulus-test-parachain-collator"))
```

Comment:
You can just call the code that generate the genesis state, instead of reading it from stdout.

Reply:
The code is in command.rs, the crate is not a lib, is there a way to import a binary like that? Or should I move this somewhere else?